src/Autocomplete/src/Controller/EntityAutocompleteController.php

```diff
@@ -54,4 +66,48 @@ public function __invoke(string $alias, Request $request): Response
             'next_page' => $nextPage,
         ]);
     }
+
+    /**
+     * @return array<string, scalar|array|null>
+     */
+    private function getExtraOptions(Request $request): array
+    {
+        if (!$request->query->has(self::EXTRA_OPTIONS)) {
+            return [];
```

Comment:
Do we still detect if anyone removes the extra options ?

Reply:
It simplifies the code. The method always returns an array, so no additional checks are required method "above".

Reply:
My scenario was: if someone manually removes the extra_options paramters in the query string, and we expect some to be defined, what would happen ?

Reply:
🤷🏼‍♂️

I believe it's not a big deal to not check whether an option was set (and I guess it might be a little bit problematic to check that), but maybe I'm too lax on this topic.

Reply:
Ok !

Reply:
What if we used Symfony's `UriSigner` instead of the checksum? If extra options exist, we sign the URL when generating it and passing it to the frontend. Here, we validate the signed URL.

Reply:
I've started reimplementing this feature, and... I noticed it computes the hash in the same way as we calculate the checksum :D.

```
private function computeHash(string $uri): string
    {
        return base64_encode(hash_hmac('sha256', $uri, $this->secret, true));
    }
```

I have troubles with testing `UriSigner` as (at least from what I know) it requires the full URL, so it's harder to simulate passing the `_hash` in the tests. If you really want to switch to `UriSigner` I'll rewrite it after the confirmation :P.